packages/components/src/menu/popover.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useContext,
+	useMemo,
+	forwardRef,
+	useCallback,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { WordPressComponentProps } from '../context';
+import type { MenuPopoverProps } from './types';
+import * as Styled from './styles';
+import { MenuContext } from './context';
+
+export const MenuPopover = forwardRef<
+	HTMLDivElement,
+	WordPressComponentProps< MenuPopoverProps, 'div', false >
+>( function MenuPopover(
+	{ gutter, children, shift, modal = true, ...otherProps },
+	ref
+) {
+	const menuContext = useContext( MenuContext );
+
+	if ( ! menuContext?.store ) {
+		throw new Error(
+			'Menu.Popover can only be rendered inside a Menu component'
+		);
+	}
+
+	// Extract the side from the applied placement — useful for animations.
+	// Using `currentPlacement` instead of `placement` to make sure that we
+	// use the final computed placement (including "flips" etc).
+	const appliedPlacementSide = Ariakit.useStoreState(
```

Comment:
Although the ESLint plugin is a bit dumb to catch this, the conditionality caused by the potential lack of `menuContext` breaks the rules of hooks. 

With that in mind, I wonder if we should just warn instead of throwing an error if there's no wrapping `<Menu />`.

Yes, there will be errors, but at least they should be preceded by the warning/error that we log.

Reply:
I do like the throwing part, though. As an alternative, we could move the line calling the hook before the check. How does that sound?

Reply:
Is moving the lines above a viable solution though? There are some other hook calls below, and some of them do need the `menuContext?.store.`

Reply:
I think so, I'll push an update soon.